target/linux/ramips/mt7620/base-files/etc/board.d/01_leds

```diff
@@ -154,6 +154,7 @@ netgear,wn3000rp-v3)
 	set_wifi_led "$boardname:green:router"
 	;;
 netgear,ex3700|\
+netgear,ex6120|\
```

Comment:
Can you check whether the device can you a DTS trigger, like used e.g. here:
https://github.com/openwrt/openwrt/blob/master/target/linux/ramips/dts/mt7620a_hiwifi_hc5861.dts#L38

Just remove the change to 01_leds and add the triggers "phy0tpt" and "phy1tpt" to the DTS above. Flash _without_ keeping settings and test on your device ...

Reply:
After making those changes (addition and removal) as indicated and reflashing the device via sysupgrade -v -n newfile.bin, there are no default LED triggers defined.

```
router_g {
			label = "ex6120:green:router";
			gpios = <&gpio2 25 GPIO_ACTIVE_LOW>;
                        linux,default-trigger = "phy0tpt";
};
```



```
device_g {
			label = "ex6120:green:device";
			gpios = <&gpio2 20 GPIO_ACTIVE_LOW>;
                        linux,default-trigger = "phy1tpt";
};
```

They can be defined manually, though.

ex6120:green:device: working
ex6120:amber:power: working
ex6120:green:power: cannot be completely off, other settings like heartbeat working
ex6120:green:router: working
ex6120:green:wps: working
ex6120:red:Device:  working
ex6120:red:router: working

Those have no visible effect and I have no clue what their effect should be - maybe they should not be offered in the dropdown list in the first place:
mt76-phy0
rt2800soc-phy1::assoc
rt2800soc-phy1::quality
rt2800soc-phy1::radio

So given the new code states "default-trigger", this is either me not being qualified to get this going, or the default trigger system not working on this type of device. Both explanations make me favour the original code that is known to work by adding default actions for two out of the four LEDs.

Reply:
I'm not exactly sure how you actually checked here, I'd just have looked whether the correct LED blinks when corresponding WiFi is active. However, for this one I can live with keeping the user-space setup and just go on to some other task.

Reply:
There is no corresponding WiFi. There is a LED labelled "Device", "WPS", "Router", "Power" each, that's it. I applied heartbeat and fully off modes and checked the device.

Reply:
I'm aware of that, but you configure the following right now (in 01_leds):
```
	ucidef_set_led_netdev "wlan5g" "ROUTER (green)" "$boardname:green:router" "wlan0"
	ucidef_set_led_netdev "wlan2g" "DEVICE (green)" "$boardname:green:device" "wlan1"
```

This means that you currently use the green:router LED for indicating activity on wlan0 and the green:device LED for indicating activity on wlan1. Assuming that this LED pattern is correct, I was just implying that the _method_ to implement it should be switched from user-space (01_leds) to device-tree by dropping the entry in 01_leds and adding the trigger in device tree.
If nothing at all is happening with these two LEDs on active WiFi for the existing configuration, then something is wrong with your GPIO assignment for the LEDs.

Reply:
Is this visible on the LED configuration page? If yes, this setting does not work and does not set up triggers. LEDs are working when configured manually, so GPIO assignment is correct. The PCB of EX6120 and EX6130 is identical, it's just a different case that offers a looped mains plug on EX6130 units.